### ISSUE TITLE
Can't use ExempleProvider.populate()

### ISSUE THREAD

**Issue by mpds** [gnina/libmolgrid#29]
I'm trying to load .pdb files using molgrid.ExampleProvider().

This is the code I wrote:
```
import molgrid
import os

thisdir = os.getcwd() + '/'
fname = thisdir + 'bind.types'

exprovider = molgrid.ExampleProvider(data_root=thisdir)
exprovider.populate(fname)
```

I'm getting this error: `ValueError: Missing molecular data in line: 1 9.278 a2weg.pdb`

My directory setup is like this:
```
|-- molvox/
    |-- a2weg.pdb
    |-- bind.types
    |-- molvox.ipynb
```

And the .types file I'm using has just one line:
`1 9.278 a2weg.pdb`

What's wrong? Thank you

**Comment by dkoes** [comment]
I cannot reproduce this.  Could your file possibly have non-printing characters in it?

**Comment by mpds** [comment]
Yes, of course. The actual files I'm using (attached in google drive because file type not supported here): https://drive.google.com/drive/folders/10EMVpGb1538uqCV7MpfQukf9HtrTUVPl?usp=sharing

**Comment by dkoes** [comment]
Still can't reproduce.  How have you installed molgrid?

![image](https://user-images.githubusercontent.com/2342142/74477805-011ee500-4e7a-11ea-85fd-670eb886084d.png)

**Comment by mpds** [comment]
Yes, I can normally import molgrid. I installed molgrid following the installation instructions in https://gnina.github.io/libmolgrid/
(basically I just used `pip install molgrid`, and after that I also had to install torch, via pip, to make it work). I also installed CUDA, cmake, and openbabel. 

I've tried to run the same code in another machine and got the same error.
![error_molgrid](https://user-images.githubusercontent.com/25796259/74479277-a3949380-4e8d-11ea-8e35-6b88adc77a35.png)

**Comment by mpds** [comment]
But I've installed openbabel and cmake via "sudo snap install..."

**Comment by dkoes** [comment]
I'm flummoxed.  I've created a fresh ubuntu docker container and still can't reproduce the error.

What operating system are you using?  Are you in a non-english locale that might have different character encodings (what's the output of `locale`)?

What happens if you do
`exprovider.populate(fname,num_labels=2)`

This error is indicating that it didn't parse the molecular filename from the line.  This could happen if it decides to interpret "a2web.pdb" as a number, which doesn't happen for me, but possibly in a different locale it is a valid thing to do.  Specifying the number of labels explicitly will prevent it from trying to determine the number automatically based on the types of the data.

**Comment by mpds** [comment]
I'm using Ubuntu 19.10. Yes, I live in Brazil. The output of `locale` is:

```
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE=pt_BR.UTF-8
LC_NUMERIC=pt_BR.UTF-8
LC_TIME=pt_BR.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=pt_BR.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=pt_BR.UTF-8
LC_NAME=pt_BR.UTF-8
LC_ADDRESS=pt_BR.UTF-8
LC_TELEPHONE=pt_BR.UTF-8
LC_MEASUREMENT=pt_BR.UTF-8
LC_IDENTIFICATION=pt_BR.UTF-8
LC_ALL=
```

I figured out something odd, when I pack up those same lines of code that are in the notebook in a single `.py` script, the script works without error! But if I try to run using a notebook, it doesn't matter if I use the flag ``num_lables=2``, I still get the same error. 

It's weird because it is exactly the same lines of code, the only difference is whether I'm executing as .py script or in a notebook (this happed in other machines I've tried). Can you imagine why could that be? I've installed jupyter notebook normally via pip.

**Comment by dkoes** [comment]
Compare what version of python jupyter is using to what is being run on the commandline.
```
import sys
print(sys.executable)
print(sys.path)
```
Are you using anaconda?

**Comment by mpds** [comment]
The output is slightly different. I'm not using anaconda.

Command line:
```
/usr/bin/python3
['/home/matheusp/python/molvox', '/usr/lib/python37.zip', '/usr/lib/python3.7', '/usr/lib/python3.7/lib-dynload', '/home/matheusp/.local/lib/python3.7/site-packages', '/usr/local/lib/python3.7/dist-packages', '/usr/lib/python3/dist-packages']
``` 

Inside jupyter notebook:
```
/usr/bin/python3
['/home/matheusp/python/molvox', '/usr/lib/python37.zip', '/usr/lib/python3.7', '/usr/lib/python3.7/lib-dynload', '', '/home/matheusp/.local/lib/python3.7/site-packages', '/usr/local/lib/python3.7/dist-packages', '/usr/lib/python3/dist-packages', '/home/matheusp/.local/lib/python3.7/site-packages/IPython/extensions', '/home/matheusp/.ipython']
```

**Comment by dkoes** [comment]
See if modifying sys.path to be equal to what you get outside of jupyter fixes the problem. Do this before importing molgrid to be on the safe side (restart kernel).

**Comment by mpds** [comment]
Even after modifying sys.path to match exactly (using just sys.path.pop() twice) and restarting, same error. I guess I'll have to try a fresh installation of everything in a conda environment.

**Comment by dkoes** [comment]
Let me know what happens - this is a mystery.

**Comment by dkoes** [comment]
It's possible this issue is due to how the module was built.  Try using this version:
https://test.pypi.org/project/molgrid/0.1.2/
and let me know if it fixes the problem.

**Comment by ziqiaoxu** [comment]
Yes. I experience the same issue if I don't import pybel from openbabel. I think pybel has to be imported before molgrid so that it can function properly.

**Comment by dkoes** [comment]
Make sure you have version 0.1.2.  There was an issue with how the package was built that could result in errors depending what order other modules were loaded.  This should be resolved in 0.1.2.  If so, please confirm so I can close the issue.  If not, please provide a reduced, reproducible test case.

**Comment by ziqiaoxu** [comment]
I just updated to 0.1.2 from PyPI, and it solved this issue. Thank you!
